### DISCUSSION TITLE
Is "podman container kill" blocking?

### DISCUSSION THREAD

**Discussion by maflcko** [containers/podman#18834] [Q&A] [answered]
I use `podman container kill --all` to clean old state and start a fresh container. However, sometimes the new container can not be created because the old one still exists. I wonder if the kill is blocking or happening in the background? Reading `podman container kill --help` didn't help.

Example output:

```
++ podman container kill --all
844b77ef5f96e3908cefd835a32c2e03f92fb187d8f2ace81faf5956e188ca46
+++ docker run --rm --detach --name my_name my_image
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
Error: creating container storage: the container name "my_name" is already in use by 844b77ef5f96e3908cefd835a32c2e03f92fb187d8f2ace81faf5956e188ca46. You have to remove that container to be able to reuse that name: that name is already in use
```

```
podman version 4.3.1

**Comment by Luap99** [comment, marked as answer]
podman kill is just sending a signal, just like the kill command it is always async.

If you want to stop containers use `podman stop` that will block until the container is actually stopped and cleaned up by podman.

**Reply by maflcko** [reply]
Thanks. I guess there is still a chance that the container refuses to stop and a kill is issued in the background, but this seems good enough for now.

**Reply by rhatdan** [reply]
You might want to look at `podman run --replace`

 podman run --help | grep replace
      --replace                                  If a container with the same name exists, replace it
